tests/bson/bug0592.phpt

```diff
@@ -0,0 +1,94 @@
+--TEST--
+PHPC-592: Property name corrupted when unserializing 64-bit integer on 32-bit platform
+--SKIPIF--
+<?php if (4 !== PHP_INT_SIZE) { die('skip Only for 32-bit platform'); } ?>
+<?php require __DIR__ . "/../utils/basic-skipif.inc"?>
+--INI--
+mongodb.debug=stderr
+--FILE--
+<?php
+
+require_once __DIR__ . "/../utils/basic.inc";
+
+$tests = [
+    '{ "x": { "$numberLong": "-2147483648" }}',
+    '{ "x": { "$numberLong": "2147483647" }}',
+    '{ "x": { "$numberLong": "4294967294" }}',
+    '{ "x": { "$numberLong": "4294967295" }}',
+    '{ "x": { "$numberLong": "9223372036854775807" }}',
+    '{ "longFieldName": { "$numberLong": "-2147483648" }}',
+    '{ "longFieldName": { "$numberLong": "2147483647" }}',
+    '{ "longFieldName": { "$numberLong": "4294967294" }}',
+    '{ "longFieldName": { "$numberLong": "4294967295" }}',
+    '{ "longFieldName": { "$numberLong": "9223372036854775807" }}',
+];
+
+foreach ($tests as $json) {
+    printf("Test %s\n", $json);
+    var_dump(toPHP(fromJSON($json)));
+    echo "\n";
+}
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+Test { "x": { "$numberLong": "-2147483648" }}
+object(stdClass)#%d (%d) {
+  ["x"]=>
+  int(-2147483648)
+}
+
+Test { "x": { "$numberLong": "2147483647" }}
+object(stdClass)#%d (%d) {
+  ["x"]=>
+  int(2147483647)
+}
+
+Test { "x": { "$numberLong": "4294967295" }}
+[%s]     PHONGO-BSON: WARNING   > ENTRY: Integer overflow detected on your platform: 4294967295
```

Comment:
I wasn't able to create a 64-bit Windows executable due to some DLL issues, so I can't confirm wjether the expected output for this test is accurate. We may need to revise.